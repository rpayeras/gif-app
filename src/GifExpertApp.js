import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifRow } from './components/GifRow';

const GifExpertApp = ({ defaultCategories = []}) => {
    const [categories, setCategory] = useState(defaultCategories)
    const onSubmitForm = value => setCategory([value, ...categories])

    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory onSubmit={onSubmitForm}/>
            <hr />
            <ol>
            {
                categories.map((category, i) => (
                    <li className="" key={'category-'+i}><GifRow category={category}/></li>
                ))
            }
            </ol>    
        </>
    );
};


export default GifExpertApp;
