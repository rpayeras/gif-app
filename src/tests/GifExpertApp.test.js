import { shallow } from 'enzyme'
import React from 'react'
import GifExpertApp from '../GifExpertApp'

describe('Testing main component', () => {
    test('should render properly', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot()
    })

    test('should show a list of categories', () => {
        const categories = ['One punch']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifRow').length).toBe(categories.length)
    })
})