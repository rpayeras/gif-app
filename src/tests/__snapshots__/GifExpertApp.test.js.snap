// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing main component should render properly 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    onSubmit={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Testing main component should show a list of categories 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    onSubmit={[Function]}
  />
  <hr />
  <ol>
    <li
      className=""
      key="category-0"
    >
      <GifRow
        category="One punch"
      />
    </li>
  </ol>
</Fragment>
`;
